app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt

```diff
@@ -540,222 +557,214 @@ fun AgendaScreenContent(
                 .tinyHeight()
         )
 
+        // • EMPTY AGENDA INDICATOR
+        val showEmptyIndicator = remember { mutableStateOf(false) }
+        LaunchedEffect(agendaItems.isEmpty()) {
+            showEmptyIndicator.value = false
+
+            if (agendaItems.isEmpty()) {
+                delay(500)
+                showEmptyIndicator.value = true
+            }
+        }
+
         // • SHOW AGENDA ITEMS LIST
-        if (agendaItems.isNotEmpty()) {
-            LazyColumn(
-                state = scrollState,
-                modifier = Modifier
-                    .background(color = MaterialTheme.colors.surface)
-                    .fillMaxSize()
-                    .padding(start = DP.tiny, end = DP.tiny)
+        LazyColumn(
+            state = scrollState,
+            modifier = Modifier
+                .background(color = MaterialTheme.colors.surface)
+                .fillMaxSize()
+                .padding(start = DP.tiny, end = DP.tiny)
 
-            ) {
-                val sortedAgendaItems = agendaItems.sortedBy { agendaItem ->
-                    agendaItem.startTime
-                }
-                val agendaItemsBeforeNow = sortedAgendaItems.filter { agendaItem ->
-                    agendaItem.startTime.isBefore(zonedDateTimeNow)
-                }
-                val agendaItemsAfterNow = sortedAgendaItems.filter { agendaItem ->
-                    agendaItem.startTime.isAfter(zonedDateTimeNow)
-                }
+        ) {
+            val sortedAgendaItems = agendaItems.sortedBy { agendaItem ->
+                agendaItem.startTime
+            }
+            val agendaItemsBeforeNow = sortedAgendaItems.filter { agendaItem ->
+                agendaItem.startTime.isBefore(zonedDateTimeNow)
+            }
+            val agendaItemsAfterNow = sortedAgendaItems.filter { agendaItem ->
+                agendaItem.startTime.isAfter(zonedDateTimeNow)
+            }
 
-                fun isToday() = (
-                        weekStartDate.plusDays(selectedDayIndex?.toLong() ?: 0).year ==
-                                ZonedDateTime.now().year
-                                && weekStartDate.plusDays(
-                            selectedDayIndex?.toLong() ?: 0
-                        ).dayOfYear ==
-                                ZonedDateTime.now().dayOfYear
-                        )
+            fun isToday() = (
+                    weekStartDate.plusDays(selectedDayIndex?.toLong() ?: 0).year ==
+                            ZonedDateTime.now().year
+                            && weekStartDate.plusDays(
+                        selectedDayIndex?.toLong() ?: 0
+                    ).dayOfYear ==
+                            ZonedDateTime.now().dayOfYear
+                    )
```

Comment:
Very hard to read for me, here I'd work with a function body and some variables because otherwise you don't really see what's going on

Reply:
ok will refactor for clarity